File: 6/1/1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {

	dirs := map[string][]int{
		"^": {-1, 0},
		">": {0, 1},
		"<": {0, -1},
		"v": {1, 0},
	}

	nextDir := map[string]string{
		"^": ">",
		">": "v",
		"<": "^",
		"v": "<",
	}

	grid := [][]string{}

	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {

		line := scanner.Text()
		grid = append(grid, strings.Split(line, ""))

	}

	total := 1

	curr := "^"
	r := 59
	c := 62

	for {
		val, _ := dirs[curr]

		x, y := val[0], val[1]
		r += x
		c += y

		if r >= 130 || r < 0 || c >= 130 || c < 0 {
			break
		}

		nextTile := grid[r][c]

		if nextTile == "." {
			grid[r][c] = "X"
			total += 1
			continue
		}

		if nextTile == "X" {
			continue
		}

		r -= x
		c -= y

		curr, _ = nextDir[curr]

	}

	fmt.Println(total)

}

File: 6/1/1.py
grid = [list(input().strip()) for _ in range(130)]

r = 59
c = 62

curr = "^"

dirs = {
    "^": (-1,0),
    ">": (0,1),
    "<": (0,-1),
    "v": (1,0)
}

next_dir = {
    "^": ">",
    ">": "v",
    "<": "^",
    "v": "<"
}

total = 1

while True:
    x, y = dirs[curr]

    r += x
    c += y

    if r >= 130 or r < 0 or c >= 130 or c < 0:
        break

    next_tile = grid[r][c]

    if next_tile == ".":
        grid[r][c] = "X"
        total += 1
        continue

    if next_tile == "X":
        continue

    r -= x
    c -= y

    curr = next_dir[curr]


print(total)
